testData/results/pkg/TestPatternMatching17FakeNew.dec

```diff
@@ -15,58 +15,37 @@ public class TestPatternMatching17FakeNew {
    }// 16
 
    void test3(Object o) {
-      if (o.hashCode() < 0 || o instanceof String s) {// 19 20
+      if (o.hashCode() < 0 || o instanceof String) {// 19
+         String s = (String)o;// 20
          System.out.println(s);// 21
       }
    }// 23
 
    void test3B(Object o) {
-      if (o.hashCode() < 0 || o instanceof String s) {// 26 27
+      if (o.hashCode() < 0 || o instanceof String) {// 26
+         String s = (String)o;// 27
          System.out.println(s);// 28
       }
 
       System.out.println("bye");// 31
    }// 32
 
    void test4(Object o) {
-      if (o.hashCode() >= 0) {// 35
-         if (!(o instanceof String)) {
-            return;// 39
-         }
-
-         String var2 = (String)o;
+      if (o.hashCode() < 0 || o instanceof String var2) {// 35
```

Comment:
Regression?

Reply:
Nope, that's what the source code. I'm not sure which part you think here regressed?

Reply:
Oh oops, I didn't have the code in front of me and presumed it was a regression. Nevermind then!